src/components/layout/Stack/Stack.tsx

```diff
@@ -36,28 +37,27 @@ export interface StackProps {
   className?: string;
 }
 
-const Stack = styled.div.attrs((props) => ({
-  ...props,
-  className: cls(CLX_LAYOUT, props?.className),
-}))<StackProps>`
+const StackStyling = css`
   display: flex;
   flex-direction: column;
   flex-wrap: nowrap;
   align-items: ${prop('justify')};
   justify-content: ${prop('align')};
+  ${parseResponsiveStyles('gap', 'gap', gapParser)};
+`;
 
-  /* FIXME: Until we remove 'margin' property from other components we need to
-    increase specificity of those nesting , since it can be overriden by inner
-    elements with the same specificity. This can lead to inconsistent output
-    of visual test if styled-components puts CSS in different order into Head. */
-  ${({ isRecursive }) => (isRecursive ? '&&' : '&& >')} * {
-    margin-top: 0;
-    margin-bottom: 0;
+const RecursiveStackStyling = css`
+  && * + * {
+    ${StackStyling}
   }
+`;
 
-  ${({ isRecursive }) => (isRecursive ? '&&' : '&& >')} * + * {
-    margin-top: ${({ gap, theme }) => getSpace(gap, { theme })};
-  }
+const Stack = styled.div.attrs((props) => ({
+  ...props,
+  className: cls(CLX_LAYOUT, props?.className),
+}))<StackProps>`
+  ${StackStyling};
+  ${({ isRecursive }) => (isRecursive ? RecursiveStackStyling : null)}
```

Comment:
Let's kill `isRecursive` property it's not used anywhere and even I didn't find any reasonable use case for it

Reply:
done ✅